tests/UriTest.php

```diff
@@ -326,6 +326,17 @@ public function testAddAndRemoveQueryValues()
         $this->assertSame('', $uri->getQuery());
     }
 
+    public function testWithQueryValues()
```

Comment:
please also add a test where existing query params are replaced like for withQueryValue